docs/gcloud-auth.rst

```diff
@@ -0,0 +1,208 @@
+.. toctree::
+  :maxdepth: 1
+  :hidden:
+
+Authentication
+--------------
+
+=======
+Summary
+=======
+
+For the majority of cases,
+authentication should "just work"
+as credentials can be loaded implicitly
+based on the environment you're application is running in
+and therefore there is no extra code required to authenticate.
+
+When running your app in production,
+you should rely on a service account.
+If you're running your application inside Google Compute Engine or
+Google App Engine,
+the service account can be automatically discovered.
+If you're running your application elsewhere,
+you should download the service account JSON keyfile
+and point to it using an environment variable.
+
+-------------------
+Order of evaluation
+-------------------
+
+``gcloud`` will look in a bunch of different places
```

Comment:
Not that isn't the order it actually happens in. See [the code][1].

The `cls._implicit_credentials_from_files` method [checks][2] first the `GOOGLE_APPLICATION_CREDENTIALS` environment variable and then the default credentials created by `gcloud auth login`.

[1]: https://github.com/google/oauth2client/blob/22a532de1d632a98beca7323a66a01fa894fbb86/oauth2client/client.py#L1159-L1163
[2]: https://github.com/google/oauth2client/blob/22a532de1d632a98beca7323a66a01fa894fbb86/oauth2client/client.py#L1120-L1122